sklearn/manifold/_spectral_embedding.py

```diff
@@ -44,6 +44,9 @@ def _graph_connected_component(graph, node_id):
         node.
     """
     n_node = graph.shape[0]
+    if isinstance(graph, np.matrix):
```

Comment:
instead, can we just pass np.asarray(graph) to _graph_connected_component, instead of passing graph unchanged?

Reply:
@jnothman OK, I did that. Also I noticed since **numpy 1.15**, it is advised not to use `np.matrix`. The following is written in the manual:

> "It is no longer recommended to use this class, even for linear algebra. Instead use regular arrays. The class may be removed in the future."